airflow/providers/hashicorp/secrets/vault.py

```diff
@@ -112,7 +112,7 @@ class VaultBackend(BaseSecretsBackend, LoggingMixin):
     def __init__(  # pylint: disable=too-many-arguments
         self,
         connections_path: str = 'connections',
-        variables_path: str = 'variables',
+        variables_path: Optional[str] = None,
```

Comment:
It is breaaking change. Is it really needed?

Reply:
@fhoda By default if Variables are not found in Vault, it will fallback to your Environment variable and then DB.

Is thing change intended to reduce the API call or any other reason?

Reply:
@mik-laj I can revert that specific part and implement is slightly differently. Where it keeps the default value that was originally there `variable_path = 'variables'`, but then user must supply `null` to that parameter when defining `backend_kwargs`. This is actually how I have done it at my company, but I was trying to match what I saw generally in Airflow for this PR.

@kaxil  It is meant to reduce API calls to Vault primarily. Vault produces errors for each task/dag that uses variable for each variable it has. We have many variables across many environments with a  large number of DAGs and prefer to manage them in the UI for the ease of the Developer. But want to keep sensitive Service Accounts and Secrets out of their hands directly, so Connections are put in Vault. Thus hitting the Vault API every time for Variables that are not going to be there is unnecessary.

Reply:
@fhoda We can change this default value, but we have to do it safely for the end user. We shoul have at least one [minor version](https://semver.org/) that has a warning about changing default value in a future version - [FutureWarning](https://docs.python.org/3/library/exceptions.html#FutureWarning). However, this can now be difficult to do if this class is in the backport provider for Airflow 1.10, provider package for Airflow 2.0. Each package has a separate release cadence, so It is not easy for us to find one version where we can inject this warning outside of the normal publishing cycle. 

It would be a lot easier if this change happened in the core. Then we can accept these changes into the 2.0 series, and then add an additional warning to the 1.10 series.

So I propose that in this release you add a warning about changing the default value in a future release. When we release Airflow 2.1 (or when there is another good time) we will introduce the breaking changes proposed by you.

Reply:
Ok, to confirm, I will revert this change to the default value.